### PR TITLE
Added missing section headers in Nautobot Version 2.0 release overvie…

### PR DESCRIPTION
**Pull request by HanlinMiao**
# Closes: #3303 
# What's Changed

Added missing section headers in the Nautobot Version 2.0 release overview and added TODO comments.

# TODO

- [x] Explanation of Change(s)
- [x] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [x] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: nautobot/docs/release-notes/version-2.0.md [hunk 1]
```diff
@@ -26,6 +30,10 @@ Added Site Model Fields to Location. Location Model now has `asn`, `comments`, `
 
 Added the `?depth` query parameter in Nautobot v2.X to replace the `?brief` parameter in the REST API. It enables [nested serialization](https://www.django-rest-framework.org/api-guide/serializers/#specifying-nested-serialization) functionality and offers a more dynamic and comprehensive browsable API. It allows users greater control of the API response data and it is available for both retrieving a single object and a list of objects. This parameter is a positive integer value that can range from 0 to 10. To learn more more, check out the [documentation on the `?depth` query parameter](../user-guide/platform-functionality/rest-api/overview.md/#depth-query-parameter).
 
+#### Added Default Nautobot API Version ([#2799](https://github.com/nautobot/nautobot/issues/2799))
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:33]: Is this redundant with "REST API Versioning Behavior" under "Changed" below?

File: nautobot/docs/release-notes/version-2.0.md [hunk 2]
```diff
@@ -52,6 +60,10 @@ Developers can refer to the [documentation on natural keys](../development/core/
 
 Two new configuration settings, [`DEVICE_NAME_AS_NATURAL_KEY`](../user-guide/administration/configuration/optional-settings.md#device_name_as_natural_key) and [`LOCATION_NAME_AS_NATURAL_KEY`](../user-guide/administration/configuration/optional-settings.md#location_name_as_natural_key), have been added to allow an administrator to customize the natural-key behavior of these two widely-used models.
 
+#### Nautobot UI 2.0 ([#3142](https://github.com/nautobot/nautobot/issues/3142))
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:63]: Probably will want to add the word "Alpha" here.

File: nautobot/docs/release-notes/version-2.0.md [hunk 3]
```diff
@@ -138,6 +150,18 @@ Check out the [Region and Site Related Data Model Migration Guide](../user-guide
 
 `nautobot.utilities` no longer exists as a separate Python module or Django app. Its functionality has been collapsed into the `nautobot.core` app. See details at [Python Code Location Changes](../user-guide/administration/upgrading/from-v1/upgrading-from-nautobot-v1.md#python-code-location-changes).
 
+#### CSV Import/Export Re-implementations ([#254](https://github.com/nautobot/nautobot/issues/254))
+
+<!-- TODO: Probably not gonna be any impact on the user end but should be worth mentioning for developers. -->
```
Comment (glennmatthews) [inline nautobot/docs/release-notes/version-2.0.md:155]: This absolutely impacts the user - the CSVs are now round-trip capable but have different structural expectations than either the "export" or "import" CSV formats in 1.x.

**Review by glennmatthews — APPROVED** [review]

**Comment by glennmatthews** [comment]
Whoops - shouldn't this be against `develop` rather than `next`?

**Comment by HanlinMiao** [comment]
> Whoops - shouldn't this be against `develop` rather than `next`?

In the story it was said to target `next` but it might be outdated. I will rebase to `develop`

File: nautobot/docs/release-notes/version-2.0.md [hunk 4]
```diff
@@ -18,6 +18,10 @@ The new Namespace model expands on the functionality previously provided by `VRF
 
 DeviceRole, RackRole, IPAM Role, and IPAddressRoleChoices have all been merged into a single generic Role model. A role can now be created and associated to one or more of the content-types that previously implemented role as a field. These model content-types include dcim.device, dcim.rack, virtualization.virtualmachine, ipam.ipaddress, ipam.prefix, and ipam.vlan.
 
+#### Added Capability to Assign IP Address to Multiple Interfaces ([#2403](https://github.com/nautobot/nautobot/issues/2403))
```
Comment (HanlinMiao) [inline nautobot/docs/release-notes/version-2.0.md:21]: https://github.com/nautobot/nautobot/issues/4488

File: nautobot/docs/release-notes/version-2.0.md [hunk 5]
```diff
@@ -52,6 +56,10 @@ Developers can refer to the [documentation on natural keys](../development/core/
 
 Two new configuration settings, [`DEVICE_NAME_AS_NATURAL_KEY`](../user-guide/administration/configuration/optional-settings.md#device_name_as_natural_key) and [`LOCATION_NAME_AS_NATURAL_KEY`](../user-guide/administration/configuration/optional-settings.md#location_name_as_natural_key), have been added to allow an administrator to customize the natural-key behavior of these two widely-used models.
 
+#### Nautobot UI 2.0 Alpha ([#3142](https://github.com/nautobot/nautobot/issues/3142))
```
Comment (HanlinMiao) [inline nautobot/docs/release-notes/version-2.0.md:59]: https://github.com/nautobot/nautobot/issues/4493

File: nautobot/docs/release-notes/version-2.0.md [hunk 6]
```diff
@@ -138,6 +146,18 @@ Check out the [Region and Site Related Data Model Migration Guide](../user-guide
 
 `nautobot.utilities` no longer exists as a separate Python module or Django app. Its functionality has been collapsed into the `nautobot.core` app. See details at [Python Code Location Changes](../user-guide/administration/upgrading/from-v1/upgrading-from-nautobot-v1.md#python-code-location-changes).
 
+#### CSV Import/Export Re-implementations ([#254](https://github.com/nautobot/nautobot/issues/254))
```
Comment (HanlinMiao) [inline nautobot/docs/release-notes/version-2.0.md:149]: https://github.com/nautobot/nautobot/issues/4490

File: nautobot/docs/release-notes/version-2.0.md [hunk 7]
```diff
@@ -138,6 +146,18 @@ Check out the [Region and Site Related Data Model Migration Guide](../user-guide
...
+
+<!-- TODO: This change is very significant and impacts both developers and users. CSVForm classes are no longer needed and export/import csv formats are altered as well. -->
+
+#### Job Overhaul ([#765](https://github.com/nautobot/nautobot/issues/765))
```
Comment (HanlinMiao) [inline nautobot/docs/release-notes/version-2.0.md:153]: https://github.com/nautobot/nautobot/issues/4489

File: nautobot/docs/release-notes/version-2.0.md [hunk 8]
```diff
@@ -138,6 +146,18 @@ Check out the [Region and Site Related Data Model Migration Guide](../user-guide
...
+
+<!-- TODO: There should be a section dedicated to Job Overhaul and some subsections detailing the differnet parts of the change in 2.0.0 Release Overview. -->
+
+#### Changed DateField to DateTimeField for ChangeLoggedModel ([#2076](https://github.com/nautobot/nautobot/issues/2076))
```
Comment (HanlinMiao) [inline nautobot/docs/release-notes/version-2.0.md:157]: https://github.com/nautobot/nautobot/issues/4491

File: nautobot/docs/release-notes/version-2.0.md [hunk 9]
```diff
@@ -314,6 +334,14 @@ Check out the specific changes documented in the table at [UI and REST API Filte
 
 Support for RQ and `django-rq`, deprecated since Nautobot 1.1.0, has been fully removed from Nautobot 2.0.
 
+#### Removed Legacy manage.py ([#1634](https://github.com/nautobot/nautobot/issues/1634))
```
Comment (HanlinMiao) [inline nautobot/docs/release-notes/version-2.0.md:337]: https://github.com/nautobot/nautobot/issues/4491

File: nautobot/docs/release-notes/version-2.0.md [hunk 10]
```diff
@@ -314,6 +334,14 @@ Check out the specific changes documented in the table at [UI and REST API Filte
...
+
+<!-- TODO: `manage.py` is something that every django app comes with. Removing it probably should be worth mentioning. -->
+
+#### Removed Slug Fields ([#2662](https://github.com/nautobot/nautobot/issues/2662))
```
Comment (HanlinMiao) [inline nautobot/docs/release-notes/version-2.0.md:341]: https://github.com/nautobot/nautobot/issues/4492

**Review by jathanism — APPROVED** [review]

**Pull request merged**